File: original_scripts/python/binary_search/gemma_2b.py
```python
def binary_search(nums, target):
    left = 0
    right = len(nums) - 1

    while left <= right:
        mid = (left + right) // 2
        guess = nums[mid]

        if guess == target:
            return mid
        elif guess < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1
```

File: original_scripts/scala/binary_search/gemma_2b.scala
```scala
def binarySearch(arr: Array[Int], target: Int): Int = {
  var low = 0
  var high = arr.length - 1

  while (low <= high) {
    val mid = (low + high) / 2
    if (arr(mid) == target) {
      return mid
    } else if (arr(mid) < target) {
      low = mid + 1
    } else {
      high = mid - 1
    }
  }

  return -1
}
```